pkg/apis/config/v1beta1/types.go

```diff
@@ -88,6 +88,12 @@ type TargetLoadPackingArgs struct {
 	DefaultRequestsMultiplier *string `json:"defaultRequestsMultiplier,omitempty"`
 	// Node target CPU Utilization for bin packing
 	TargetUtilization *int64 `json:"targetUtilization,omitempty"`
+	// Types of client to be used: k8s, prometheus, load-watcher
+	MetricProviderType *string `json:"metricProviderType,omitempty"`
+	// Address of metric endpoint
+	MetricProviderAddress *string `json:"metricProviderAddress,omitempty"`
+	// Authentication token for metric client
+	MetricProviderToken *string `json:"metricProviderToken,omitempty"`
```

Comment:
Probably it makes more sense to define it as:

```yaml
metricProvider:
    type: ...
    address: ...
    token: ...
```

Reply:
Yes, I changed to it accordingly.